reflection.md

```diff
@@ -1,9 +1,20 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
-
+||, ~, <*>, <+> were all rather easy to implement because they are all pretty
+much regular functions in scala.  The literal extension for characters was very
+straight forward and was also pretty easy to implement.
 ## Which operators were most difficult to implement and why?
+I couldn't figure out how to implement the {n}
+function because it seems like it doesn't have a name.  It looks perhaps like I
+could've done something with a curried function which is the way we did some of
+the control flow stuff, however there is no name that comes before it either.
+Hmmm... now I"m wondering if maybe I could've curried the constructor of a literal
+or something like that but I'm doubtful. 
 
```

Comment:
I also struggled with this one.  I found the answer by searching on Google something like "overload parentheses Scala", and I found that if you define a function `apply` then it works as you want it to.  I think it will work if you define an `apply` function method in the RegularExpression class, because it will convert '3' into a RegularExpression and then match with the apply operator.